mangaki/mangaki/views.py

```diff
@@ -614,20 +614,23 @@ def get_works(request, category):
     return HttpResponse(json.dumps(data), content_type='application/json')
 
 
-def get_reco_list(request, category, editor):
+@login_required
+def get_reco_algo_list(request, algo, category):
     reco_list = []
-    for work, is_manga, in_willsee in get_recommendations(request.user, category, editor):
-        reco_list.append({'id': work.id, 'title': work.title, 'poster': work.ext_poster, 'synopsis': work.synopsis,
-            'category': 'manga' if is_manga else 'anime', 'rating': 'willsee' if in_willsee else None})
-    return HttpResponse(json.dumps(reco_list), content_type='application/json')
-
+    data = get_reco_algo(request.user, algo, category)
+    works = data['works']
+    for work_id in data['work_ids']:
+        work = works[work_id]
+        reco_list.append({'id': work.id, 'title': work.title, 'poster': work.ext_poster, 'synopsis': work.synopsis, 'category': work.category.slug})
+    return HttpResponse(json.dumps(reco_list, indent=4), content_type='application/json')
```

Comment:
Wait, I still don't understand why `indent=4`. Whether you use `indent=4` or not will only change the prettiness of the JSON output.

Also, AFAIK, dicts are not guaranteed to maintain insertion order (that's a CPython implementation detail, currently.)

If you need order, add a list in the JSON response like: `reco_ids: [1, 5, 7, 2, 3]` that is guaranteed to maintain order.
If you really need an ordered dict, you can use `collections.OrderedDict`.

(though: I wonder how would compare a list solution over an OrderedDict solution in terms of performance, that'd be definitely minor and would save up some bytes over wire. I'd say that OrderedDict might be the way to go here, what do you think @Elarnon ?)